camel/utils/commons.py

```diff
@@ -241,3 +182,81 @@ def check_server_running(server_url: str) -> bool:
 
     # if the port is open, the result should be 0.
     return result == 0
+
+
+def get_openai_function_schema(func: Callable) -> Dict[str, any]:
+
+    def _remove_a_key(d, remove_key) -> None:
+        r"""Remove a key from a dictionary recursively"""
+        if isinstance(d, dict):
+            for key in list(d.keys()):
+                if key == remove_key and "type" in d.keys():
+                    del d[key]
+                else:
+                    _remove_a_key(d[key], remove_key)
+
+    parameters: Mapping[str, Parameter] = signature(func).parameters
+    raw_function = func
+    arg_mapping: Dict[int, str] = {}
+    positional_only_args: Set[str] = set()
+    type_hints = _typing_extra.get_type_hints(func, include_extras=True)
+    fields: Dict[str, Tuple[Any, Any]] = {}
+    for i, (name, p) in enumerate(parameters.items()):
+        if p.annotation is p.empty:
+            annotation = Any
+        else:
+            annotation = type_hints[name]
+
+        default = ... if p.default is p.empty else p.default
+        if p.kind == Parameter.POSITIONAL_ONLY:
+            arg_mapping[i] = name
+            fields[name] = annotation, default
+            positional_only_args.add(name)
+        elif p.kind == Parameter.POSITIONAL_OR_KEYWORD:
+            arg_mapping[i] = name
+            fields[name] = annotation, default
+
+        elif p.kind == Parameter.KEYWORD_ONLY:
+            fields[name] = annotation, default
+        elif p.kind == Parameter.VAR_POSITIONAL:
+            fields[name] = Tuple[annotation, ...], None
+        else:
+            assert p.kind == Parameter.VAR_KEYWORD, p.kind
+            fields[name] = Dict[str, annotation], None
+    model = create_model(to_pascal(raw_function.__name__), **fields)
+
+    parameters = model.model_json_schema()
+
+    parameters["properties"] = {
+        k: v
+        for k, v in parameters["properties"].items()
+        if k not in ("args", "kwargs")
+    }
+    _remove_a_key(parameters, "additionalProperties")
```

Comment:
It may be better to use abosulte dict paths instead of recursives? I'm not sure.

Reply:
I referenced the implementation in the ·instructor` library because the keyword to be removed may exist at different levels in the dictionary. For example, the location of 'title' can be:
["type"]["parameters"]["$defs"] ["type"]["parameters"]["title"]
["type"]["parameters"]["$defs"] ["function"]["parameters"]["$defs"]["title"]
["type"]["parameters"]["$defs"] ["function"]["parameters"]["properties"]["str_para"]["title"]
["type"]["parameters"]["$defs"] ["function"]["parameters"]["properties"]["int_para"]["title"]